blocks/main_loop.py

```diff
@@ -94,6 +178,8 @@ def __init__(self, algorithm, data_stream,
         self.log = log
         self.extensions = extensions
 
+        self.timer = Timer()
```

Comment:
I am slightly unhappy that the timing information is not a part of the log. On the other hand thinking about putting it here with all this uncertainty we have in #476 is rather hard. Thoughts?